crates/bdk/src/wallet/mod.rs

```diff
@@ -159,14 +160,32 @@ impl Wallet {
             NewError::Persist(_) => unreachable!("no persistence so it can't fail"),
         })
     }
+
+    /// Infallibly return a derived address using the external descriptor, see [`AddressIndex`] for
+    /// available address index selection strategies. If none of the keys in the descriptor are derivable
+    /// (i.e. does not end with /*) then the same address will always be returned for any [`AddressIndex`].
+    pub fn get_address(&mut self, address_index: AddressIndex) -> AddressInfo {
+        self.try_get_address(address_index).unwrap()
+    }
```

Comment:
This is implemented on `Wallet<()>` but it should be implemented on `Wallet<D> where D: PersistBackend<WriteError=core::convert::Infallible>`

Reply:
If I do this it looks like I also have to add a `db: D` parameter to `new_no_persist()`, and then all caller (tests for now) need to specify a `db: ()`. Is there scenario you're thinking of where D is some other `Infallible` database?

Reply:
I think Lloyd meant that you should keep `new_no_persist` in `impl Wallet`, but move the other two methods in `impl Wallet<D: ...>`:
```diff
diff --git a/crates/bdk/src/wallet/mod.rs b/crates/bdk/src/wallet/mod.rs
index 051f9730..061a3db5 100644
--- a/crates/bdk/src/wallet/mod.rs
+++ b/crates/bdk/src/wallet/mod.rs
@@ -226,7 +226,12 @@ impl Wallet {
             NewError::Persist(_) => unreachable!("no persistence so it can't fail"),
         })
     }
+}
 
+impl<D> Wallet<D>
+where
+    D: PersistBackend<ChangeSet, WriteError = core::convert::Infallible>,
+{
     /// Infallibly return a derived address using the external descriptor, see [`AddressIndex`] for
     /// available address index selection strategies. If none of the keys in the descriptor are derivable
     /// (i.e. does not end with /*) then the same address will always be returned for any [`AddressIndex`].

```

To me, it makes sense to have `get_address` and `get_internal_address` for all the wallets that have infallible backends (which could be either `()` or some other database)

Reply:
Fixed!